### PR TITLE
local routes with scope host

### PR DESCRIPTION
**Pull request by tomastigera**
This allows routes like

ip route add local 0.0.0.0/0 dev lo table 100

## Todos
- [ ] Unit tests (full coverage)
- [ ] Integration tests (delete as appropriate) In plan/Not needed/Done
- [ ] Documentation
- [ ] Backport
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

File: ifacemonitor/update_filter.go [hunk 1]
```diff
@@ -121,6 +121,10 @@ mainLoop:
 				logrus.WithField("route", routeUpd).Debug("Ignoring route with no link index.")
 				continue
 			}
+			if routeUpd.Dst == nil {
```
Comment (fasaxc) [inline ifacemonitor/update_filter.go:124]: Would be good to UT this case

**Comment by tomastigera** [comment]
This hasn't died yet, I will get to asap

**Comment by caseydavenport** [comment]
@tomastigera is this still in progress?

**Comment by caseydavenport** [comment]
The Calico team is reorganizing our code and repository structure in order to make development easier, and as such all of the code in this repository is now located at https://github.com/projectcalico/calico. We appreciate your PR and your patience as we work to improve our developer experience, and welcome you to re-open your changes against the new location. Please reach out if you have any questions!

**Pull request closed without merging**